app.js

```diff
@@ -108,44 +131,71 @@ router.post('/account/:phoneNumber/:accountId/requestCode', async ctx => {
 const nacl = require('tweetnacl');
 const crypto = require('crypto');
 const bs58 = require('bs58');
-const verifySignature = async (nearAccount, securityCode, signature) => {
-    const hasher = crypto.createHash('sha256');
-    hasher.update(securityCode);
-    const hash = hasher.digest();
-    const helperPublicKey = (await keyStore.getKey(recoveryKeyJson.account_id)).publicKey;
-    const accessKeys = await nearAccount.getAccessKeys();
-    if (!accessKeys.find(it => it.public_key == helperPublicKey.toString())) {
-        throw Error(`Account ${nearAccount.accountId} doesn't have helper key`);
+const verifySignature = async (nearAccount, data, signedData) => {
```

Comment:
Technically this is `signature` only, not `signedData` (same goes for `blockNumberSigned` – it's really `blockNumberSignature`). That's why `nacl.sign.detached` is used.  There is also variant which actually operates on signed data (https://github.com/dchest/tweetnacl-js/blob/master/README.md#naclsignopensignedmessage-publickey).  So I'm not sure why you decided to rename.

Reply:
Thanks for clarifying! I'll update the variable names.